### ISSUE TITLE
Question on CC2420

### ISSUE THREAD

**Issue by federicodotta** [travisgoodspeed/goodfet#50]
Hi,

I have an ApiMote that probably has one revision of the CC2420 with the problem you mention at line 170 of file /firmware/apps/radios/ccspi.c:

```
This_ reads too far on some CC2420 revisions, but on others it works fine.  It probably has to do with whether FIFO drops before or after the SPI clocking. 

software fix is to reset the CC2420 between packets.  This works, but a better solution is desired.
```
 
How can I add this reset command to fix the issue?

Thank you for you support!

Federico

**Comment by federicodotta** [comment]
I close the issue. My problem was caused by an issue of the recent version of goodfet fixed by #51 

Now my ApiMote works correctly.

Thank you,
Federico

**Comment by rmspeers** [comment]
@travisgoodspeed I took a look at this but not sure where it's helping - seems like to apply we'd want to have a good feeling of when this is firing and that it's helping, e.g., tracing statements. I'm also thinking the best way to handle packet corruption may be to do a ring buffer in FW to move data off of the CC2420 quicker and into the MSP chip to wait for the app layer to request?
